compat.h

```diff
@@ -29,8 +29,10 @@
 #include <string.h>
 
 #ifdef _WIN32
-#include <BaseTsd.h>
-typedef SSIZE_T ssize_t;
+#ifndef _SSIZE_T_DEFINED
+#undef ssize_t
```

Comment:
if is not defined why do undef?

Reply:
It is an extra layer of protection. If _SSIZE_T_DEFINED is not defined, it doesn't necessarily mean that ssize_t is not defined either.